transports/quic/src/provider/async_std.rs

```diff
@@ -59,4 +60,16 @@ impl super::Provider for Provider {
     fn sleep(duration: Duration) -> BoxFuture<'static, ()> {
         async_std::task::sleep(duration).boxed()
     }
+
+    fn send_to<'a>(
+        udp_socket: &'a UdpSocket,
+        buf: &'a [u8],
+        target: std::net::SocketAddr,
+    ) -> BoxFuture<'a, io::Result<usize>> {
+        Box::pin(async move {
+            async_std::net::UdpSocket::from(udp_socket.try_clone()?)
```

Comment:
When does `try_clone` fail? The docs don't mention anything.

Reply:
My understanding:

- A `UdpSocket` boils down to a file descriptor. See e.g. https://doc.rust-lang.org/src/std/sys/unix/net.rs.html#30
- "Duplicating" (i.e. cloning) a file descriptor can fail due to various reasons, e.g. the underlying fail being closed or the process is running out of file descriptors.

I am assuming that you wonder whether we can do without propagating the error. Given that we already need a way to propagate the error from `send_to`, I don't think we should apply any special treatment to the error from `try_clone`. Are you fine with that @thomaseizinger?

Reply:
Sounds good to me! I was just wondering whether there are any specific error cases :)

Happy to keep as is!